lib/compress/index.js

```diff
@@ -3915,9 +3915,11 @@ merge(Compressor.prototype, {
             if (vars_found > 0) {
                 // collect only vars which don't show up in self's arguments list
                 var defs = [];
-                vars.forEach(function(def, name) {
-                    if (self instanceof AST_Lambda
-                        && self.args_as_names().some((x) => x.name === def.name.name)) {
+                const args_as_names = self.args_as_names();
+                const isAstLambda = self instanceof AST_Lambda;
```

Comment:
Can you get rid of this check? It's redundant. This entire method should be on AST_Lambda anyway.

I was going to nitpick about the camelCase but the variable is not necessary in the first place.

Reply:
Done